import_role - support templating _from_files with --extra-vars (#75269)
ansible/ansible: docs/docsite/rst/user_guide/playbooks_reuse.rst

Importing incorporates playbooks in other playbooks statically. Ansible runs the plays and tasks in each imported playbook in the order they are listed, just as if they had been defined directly in the main playbook.

> Imported playbook filenames can use variables defined with ``--extra-vars`` or the ``vars`` keyword:

Reviewer: I'd make this more user-focused than feature-focused. Something like:

```suggestion
You can select which playbook you want to import at runtime by defining your imported playbook filename with a variable, then passing the variable with either ``--extra-vars`` or the ``vars`` keyword. For example:
```

.. code-block:: yaml

>    - import_playbook: "/path/to/{{ extra_var }}"

Reviewer: Hm, this took a few reads for me to parse. How about:
```suggestion
   - import_playbook: "/path/to/{{ import_from_extra_var }}"
```

   - import_playbook: "{{ playbook }}"
     vars:

>        playbook: /path/to/playbook.yml

Reviewer: If we followed the pattern, and showed the other option for defining the path, maybe:
```suggestion
   - import_playbook: "{{ import_from_vars }}"
     vars:
       import_from_vars: /path/to/one_playbook.yml

If you run this playbook with ``ansible-playbook my_playbook -e import_from_extra_var=other_playbook.yml``, Ansible imports both one_playbook.yml and other_playbook.yml.
```

You can still use the bare :ref:`roles <roles_keyword>` keyword at the play level to incorporate a role in a playbook statically. However, the bare :ref:`include <include_module>` keyword, once used for both task files and playbook-level includes, is now deprecated.

> Included and imported task files and roles support templating filenames. ``import_*`` actions need the variables to be available at playbook parse time (which can be done by defining them with ``--extra-vars``, for example), whereas ``include_*`` actions need the variables at runtime.

Reviewer: Since there are differences in how this functionality works for includes and imports, I'd put this information under the separate sections further down.

Author: Good idea. I moved this into the separate sections in 34df5ccf43f2178a8222a021ecb9b2177b316da6.
